### DISCUSSION TITLE
How do the program know when Monika is leaning?

### DISCUSSION THREAD

**Discussion by Lu1508** [Monika-After-Story/MonikaModDev#9391] [QA - code and content creation (technical)]
As I said before, I'm working in interactive buttons to be able to touch her (asking her before or just surprising her)

However, my BIGGEST problem is the lean pose, because that change the position of almost every part of her

I tried "if lean:", others variables and def, but is not working

I know I could try to "frozen" her in a static pose, but I don't want to do that, that will make the interaction less realistic

Is there a way to know when the actual pose of Monika is leaning and when inmediately stop?

**Comment by howltrek** [comment]
I'm not sure it can, but if the mod were based on how the clothing sprites change, with Monika's pose, it should be able to activate a different code, when she adopts that pose.

**Reply by Lu1508** [reply]
that's what I think too, but the code is so big and complex that I did find a lot of things (like m 5dsbla, 5lean, NUM_POSE[5], L_POSES, etc), but none of them could be used as variable for "if"

**Reply by howltrek** [reply]
the clothes occupy a scheme with only 7 poses.

we need someone to know how to use those poses in your submod.

({
    "__author": "mi_fan_arts",
    "version": 3,
    "type": 2,
    "name": "mi-fan-arts-succubus",
    "img_sit": "mi-fan-arts-succubus",
    "pose_map": {
        "mpm_type": 0,
        "p1": true,
        "p2": true,
        "p3": true,
        "p4": true,
        "p5": true,
        "p6": true,
        "p7": true
    },
    "stay_on_start": true,
    "ex_props": {
        "costume": "o31",
        "cosplay": true,
        "lingerie": true,
        "succubus_roleplay":true
    },
    "select_info": {
        "display_name": "succubus costume",
        "thumb": "mi-fan-arts-succubus",
        "group": "clothes",
        "visible_when_locked": true,
        "hover_dlg": [
            "Come slowly Eden Lips",
            "unused to thee",
            "Bashful sip thy jasmines",
            "As the fainting bee."
        ],
        "select_dlg": [
            "What fantasies are you planning?"
        ]
    },
    "pose_arms": {
        "crossed": {
            "tag": "crossed",
            "layers": "5^10"
        },
        "left-down": {
            "tag": "down",
            "layers": "0"
        },
        "left-rest": {
            "tag": "rest",
            "layers": "10"
        },
        "right-down": {
            "tag": "down",
            "layers": "0"
        },
        "right-point": {
            "tag": "point",
            "layers": "0"
        },
        "right-restpoint": {
            "tag": "restpoint",
            "layers": "10"
        },
        "steepling": {
            "tag": "steepling",
            "layers": "10"
        },
        "def|left-def": {
            "tag": "def",
            "layers": "10"
        },
        "def|right-def": {
            "tag": "def",
            "layers": "5^10"
        }
    },
    "giftname": "succubus_cosplay"
})

**Reply by Lu1508** [reply]
maybe one thing that could work is adding an "xpos" and "ypos" inside each pose of the "pose_map" and "pose_arms", but I'm not sure how to connect that to an imagebutton

we need someone how know to solve this issue indeed, but at least the beta version with the static Monika is finished

**Comment by Booplicate** [comment]
Are you using #4917?

**Reply by ThePotatoGuy** [reply]
^also should add, clickzones only work for non-lean, but the system can support lean if we have vertices for the lean pose at default zoom.

**Reply by Lu1508** [reply]
no, I didn't use it, I just used some invisible imagebuttons that appear when you are in "Touch" zone (if you want to see it, the code is in extradialogue.rpy in https://drive.google.com/drive/folders/1vCNBj91FI4fbCJe0-ghLB2c6LSJ7ZQOT?usp=sharing)

I'm gonna check the clickzones and see how to use it (however, I'm a newbie in renpy language, so let's see what I can do ^^")

**Reply by howltrek** [reply]
https://github.com/Monika-After-Story/MonikaModDev/wiki/Adding-Sprite-Objects#prog-points

**Reply by Booplicate** [reply]
If you're not using the clickzones framework, then you can use #8365 to fetch current exp. I'd suggest to look into clickzones, though.
